src/animalShelter/AnimalShelter.ts

```diff
@@ -72,19 +74,15 @@ export default class AnimalShelter {
     };
   };
 
-  private hasMethod(animal: Animal, methodName: string): boolean {
-    return typeof (animal as any)[methodName] === "function";
-  };
-
   public report() {
     let animalsWhoJump = 0;
     let animalsWhoRun = 0;
     let anaimalsWhoClimbTrees = 0
     for (const animal of this.animals) {
-      if (this.hasMethod(animal, "jump")) {
+      if ((animal as unknown as ICanJump).jump) {
```

Comment:
casting to unknown and then to ICanJump is also not a great solution. If you saw in the group chat Rosen posted a solution: in the `hasMethod` method you can `return animal.constructor.property.hasOwnProperty(methodName)` I believe that works but test it yourself too